server/datastore/mysql/software.go

```diff
@@ -430,92 +512,94 @@ func selectSoftwareSQL(hostID *uint, opts fleet.SoftwareListOptions) (string, []
 			ds = ds.Where(goqu.I("shc.team_id").Eq(0))
 		}
 	}
+
+	ds = ds.GroupBy(
+		"s.id",
+		"scv.cpe_id",
+		"generated_cpe",
+	)
+
+	// Pagination is a bit more complex here due to left join with software_cve table and aggregated columns from cves table.
+	// Apply order by again after joining on sub query
 	ds = appendListOptionsToSelect(ds, opts.ListOptions)
 
+	// join on software_cve and cves after pagination using a sub-query
```

Comment:
Nit: `// ... after pagination using the sub-query above.`

Reply:
fixed in 67d0e1ca9